### ISSUE TITLE
2 New Chat Bugs + Extrap bug

### ISSUE THREAD

**Issue by pixelhaxballer** [haxball/haxball-issues#1939]
**1:** Ever since the chat update, there's a very annoying bug. Sometimes the chat just refuses to keep auto scrolling down with the chat as new messages are input. It takes many tries of scrolling the chat down with either mouse wheel or dragging down the chat with the side bar hoping it eventually locks in and keeps auto scrolling. 

It's great when you're trying to scroll up and read messages, its just very very hard to get it to lock back in when you scroll to the bottom sometimes and I cannot figure out what makes it lock back in. It just seems like pure luck.

**2:** Also since the new chat updates, when you try to highlight messages to copy them, its scrolls up WAY too fast as you're trying to go up. See video for example: https://streamable.com/4mwgps

**3:** If you put your extrapolation above 200, it resets to 200 when you join a new room. I'm assuming its with anything above 200, I just noticed it when i set my extrap to 1000 to test something

**Comment by pixelhaxballer** [comment]
Also, I think scrolling through chat is a bit too slow now compared to before. It's nice and smooth just kinda slow.

**Comment by basro** [comment]
The extrapolation bug is now fixed.

**Comment by basro** [comment]
Regarding the chat not scrolling down, I've not been able to reproduce it yet.

Just in case I'd like explain an intended behavior the chat log has: The chat will not automatically scroll down if you have your mouse pointer hovering on top of it. This is there so that you are able to scroll the log up and stay up while people are still chatting.

Is there any chance that you had your mouse hovering the log when you were having the chat not scrolling down issues?

Regarding the difference in scrolling speed: This is likely because I've gotten rid of a custom JS scrollbar and changed to native browser scrollbars. I'm most likely not bringing that back since I'd much rather use native html features if possible.

**Comment by kingjasko** [comment]
the chat is definitely bugged (refusing to scroll down quite often when hiding/unhiding it), i will try to replicate and share my findings here

**Comment by kingjasko** [comment]
https://youtu.be/xGXd_fg_dn8 @basro I've enabled chat multiple times during this match and only got the issue once, next time i did it at the end of the game, it was working normally, thoughts?

**Comment by basro** [comment]
Questions come to mind:
Can it be reproduced without the extension?
Can it be reproduced without unlocked fps?
Does it ever fix itself? what happens if you chat?

**Comment by kingjasko** [comment]
NO, I have NOT managed to reproduce the bug without the extension, only WITH it (all in one extension)
Yes
Yes, if someone types something after the bug has happened, the chat will then appear scrolled all the way down, as intended.

The bug I'm noticing seems to be happening when you click "Esc" while the chat is hidden (apparently only when chat isn't first visible normally when the game starts), enabling it the next time often causes it to be scrolled all the way up. 

I also noticed chat focus height does absolutely nothing for some reason (nothing to do with the extension), neither in compact nor full mode. I'm pretty sure it worked when you first introduced it, but something must've happened and it's not due to low latency canvas. The toggle chat button sometimes stops working as well, I have not been able to reliably reproduce it, though.

https://www.youtube.com/watch?v=C1AarTT6P5k

**Comment by basro** [comment]
> NO, I have NOT managed to reproduce the bug without the extension, only WITH it (all in one extension)

This is probably the case but just to be sure, you mean you tried but failed to reproduce it without the extension, right?
This is a bit troubling, the solution to the bug is probably to fix the extension if that is the case.

> I also noticed chat focus height does absolutely nothing for some reason (nothing to do with the extension), neither in compact nor full mode. I'm pretty sure it worked when you first introduced it, but something must've happened and it's not due to low latency canvas.

Chat focus height is working normally for me. Keep in mind that it wont do anything if the focus height is lower than your current chat height. Try setting it to it's max value to see if nothing happens.

**Comment by kingjasko** [comment]
I tried it towards the end of the video, you can see it not working, am I doing something wrong?

**Comment by kingjasko** [comment]
> This is probably the case but just to be sure, you mean you tried but failed to reproduce it without the extension, right?

Yes, the issue comes down to the all-in-one tool, disabling it fixes the bug.
